File: Assets/zfoolua/http/HttpHelloResponse.lua

local HttpHelloResponse = {}

function HttpHelloResponse:new()
    local obj = {
        message = "" -- string
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function HttpHelloResponse:protocolId()
    return 1701
end

function HttpHelloResponse:write(buffer, packet)
    if packet == nil then
        buffer:writeInt(0)
        return
    end
    buffer:writeInt(-1)
    buffer:writeString(packet.message)
end

function HttpHelloResponse:read(buffer)
    local length = buffer:readInt()
    if length == 0 then
        return nil
    end
    local beforeReadIndex = buffer:getReadOffset()
    local packet = HttpHelloResponse:new()
    local result0 = buffer:readString()
    packet.message = result0
    if length > 0 then
        buffer:setReadOffset(beforeReadIndex + length)
    end
    return packet
end

return HttpHelloResponse

File: Assets/zfoocs/Http/HttpHelloResponse.cs
using System;
using System.Collections.Generic;
namespace zfoocs
{
    
    public class HttpHelloResponse
    {
        public string message;
    }

    public class HttpHelloResponseRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 1701;
        }
    
        public void Write(ByteBuffer buffer, object packet)
        {
            if (packet == null)
            {
                buffer.WriteInt(0);
                return;
            }
            HttpHelloResponse message = (HttpHelloResponse) packet;
            buffer.WriteInt(-1);
            buffer.WriteString(message.message);
        }
    
        public object Read(ByteBuffer buffer)
        {
            int length = buffer.ReadInt();
            if (length == 0)
            {
                return null;
            }
            int beforeReadIndex = buffer.GetReadOffset();
            HttpHelloResponse packet = new HttpHelloResponse();
            string result0 = buffer.ReadString();
            packet.message = result0;
            if (length > 0)
            {
                buffer.SetReadOffset(beforeReadIndex + length);
            }
            return packet;
        }
    }
}